### ISSUE TITLE
a tag nesting breaks everything

### ISSUE THREAD

**Issue by lehtu** [taoqf/node-html-parser#211]
It seems that this lib cannot handle `a` tag nesting. If there is an `a` tag inside another one and there are other tag openings in between, the parser will force close all tags between two `a` tag openings.

This is of course not valid HTML, although it's semantically valid, but still I thought that parser would do what it promises. As a developer I want to be able to parse HTML no matter who has produced it, especially if it's semantically alright.

Is there any workarounds for this?

example1:
```js
import { parse } from 'node-html-parser';

const root = parse(`
  <a>
    <table>
      <tr>
        <td>
          <a href="https://github.com">link to github</a>
        </td>
      </tr>
    </table>
  </a>
`);

console.log(root.toString());
// outputs
<a>
<table>
  <tr>
    <td>
      </td></tr></table></a><a href="https://github.com">link to github</a>
```

example2:
```js
import { parse } from 'node-html-parser';

const root = parse(`
    <a>
      <a href="https://github.com">link to github</a>
    </a>
`);

console.log(root.toString());
//outputs:
    <a>
      </a><a href="https://github.com">link to github</a>
```

**Comment by nonara** [comment]
Hi @lehtu (and @taoqf)

I responded on another thread about this, but I'll update here, as this seems to be the root thread on the issue.

I had a look into other parsers, and it seems that there are two ways that this is handled. It can be handled as-is, despite being invalid HTML, or it can be corrected, as we are doing since #148

Most parsers do allow the broken html, but notably `parse5` does not, because it's parsing in a way that produces an HTML document. Additionally, if you insert the same code into HTML in Chrome debugger, you'll see that it corrects them as well.

Ultimately, I think that in our specific scenario, the former behaviour (which you've suggested) is the right course of action. It would really only make sense if the point of the parser was to produce a fully compliant HTML spec, with corrections to all issues.

As a result, we can fix it by reverting changes made in #148, however, this is a breaking change. With that in mind, I will add a legacy option to perform the A tag fix (as some code relies on it), and we'll release as a new major version.

I will handle this on Sunday.

**Comment by nonara** [comment]
Closed by #215 

Released in `v6.0.0`
